packages/codegen/src/templates/library/imports-bridge.ts

```diff
@@ -110,14 +111,14 @@ function wrapJSIReturnIntoNative(
   // Handle multiple value types (struct)
   if (resultTypes.length > 1) {
     const elements = resultTypes
-      .map((t, i) => toJSINumber(`${varName}.${STRUCT_TYPE_PREFIX[t]}${i}`))
+      .map((t, i) => toJSINumber(`${varName}.${STRUCT_TYPE_PREFIX[t]}${i}`, t))
       .join(', ');
 
     return `return { ${elements} }`;
   }
 
   if (resultTypes.length === 1) {
-    return `return coerceToNumber<${func.returnTypeName}>(res)`;
+    return fromJSINumber('res', resultTypes[0]!, func.returnTypeName);
```

Comment:
This is currently not getting returned, which breaks things.

```suggestion
    return `return ${fromJSINumber('res', resultTypes[0]!, func.returnTypeName)}`;
```